src/network/MissionControlProtocol.cpp

```diff
@@ -121,6 +127,33 @@ void MissionControlProtocol::handleDriveRequest(const json& j) {
 	this->setRequestedCmdVel(dtheta, dx);
 }
 
+void MissionControlProtocol::handleSetArmIKEnabled(const json& j) {
+	bool enabled = j["enabled"];
+	Globals::armIKEnabled = enabled;
```

Comment:
In the `true` case we set it to false and then back to true again, so this line is extraneous. It would be cleaner to just set `Globals::armIKEnabled = false` in the false case (before the `is_joinable` call) and delete this line.